### PR TITLE
Add bcftools view module

### PR DESCRIPTION
**Pull request by abhi18av**
## PR checklist

This PR adds `bcftools/view` module.

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `<SOFTWARE>.version.txt` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

**Review by jfy133 — CHANGES_REQUESTED** [review]
When checking the `--help`, it seems that there are additional optional input files - namely `-R`, `-T`, `-S`.

You should define input channels for these, and add conditions that insert them into the command if supplied.

**Comment by abhi18av** [comment]
Thanks for the review @jfy133 ! 

Yes, that's a good catch. Could you please point me to another module which makes use of optional input? I'm not particularly sure how to do this 🤔

**Comment by jfy133** [comment]
1) Example: 
  - https://github.com/nf-core/modules/blob/ab93a1afa7941dcf4fff8dc35712290077f25413/modules/malt/build/main.nf#L26
  - https://github.com/nf-core/modules/blob/ab93a1afa7941dcf4fff8dc35712290077f25413/modules/malt/build/main.nf#L42
  - https://github.com/nf-core/modules/blob/ab93a1afa7941dcf4fff8dc35712290077f25413/modules/malt/build/main.nf#L50
  - https://github.com/nf-core/modules/blob/ab93a1afa7941dcf4fff8dc35712290077f25413/tests/modules/malt/build_test/main.nf#L11
  
  2) You would need to make some dummy files and add to `genomes/` in test-dataset (even if a defininig a single region), I don't work much with VCF nor bcftools so I can't give more concrete help there (sorry)

File: modules/bcftools/view/meta.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    - meta:
+        type: map
+        description: |
+          Groovy Map containing sample information
+          e.g. [ id:'test', single_end:false ]
+    - vcf:
+        type: file
+        description: |
+          The vcf file to be inspected.
+          e.g. 'file1.vcf'
```
Comment (jfy133) [inline modules/bcftools/view/meta.yml:26]: Missing the optional input files

**Review by jfy133 — APPROVED** [review]
Missing some documentation of the optional input files, then we are good to go.

I think it's OK to add the test for sample files in the future for the time being, as it's exactly the same format as the `targets_file` condition.

File: modules/bcftools/view/meta.yml (hunk 1 above)
Comment (abhi18av) [reply modules/bcftools/view/meta.yml]: I've addressed these via https://github.com/nf-core/modules/pull/669/commits/77769e3b93c489aca19c403a0555eb0a2584c9bd

**Pull request merged**